src/_includes/layouts/base.njk

```diff
@@ -5,6 +5,27 @@
     <title>{% if '/' === page.url %}V8 JavaScript engine{% else %}{{ (renderData.title or title) | markdown | striptags | decodeHtmlEntities }} · V8{% endif %}</title>
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <meta name="color-scheme" content="dark light">
+    <script>
+      // If `prefers-color-scheme` is not supported, fall back to light mode.
+      // In this case, light.css will be downloaded with `highest` priority.
+      if (window.matchMedia('(prefers-color-scheme: dark)').media === 'not all') {
+        document.documentElement.style.display = 'none';
+        document.head.insertAdjacentHTML(
+            'beforeend',
+            '<link rel="stylesheet" href="/_css/light.css" onload="document.documentElement.style.display = \'\'">'
+        );
+      }
+    </script>
+    <!--
+      Conditionally either load the light or the dark stylesheet. The matching file
+      will be downloaded with `highest`, the non-matching file with `lowest`
+      priority. If the browser doesn't support `prefers-color-scheme`, the media
+      query is unknown and the files are downloaded with `lowest` priority (but
+      above I already force `highest` priority for my default light experience).
+    -->
+    <link rel="stylesheet" href="/_css/dark.css" media="(prefers-color-scheme: dark)">
+    <link rel="stylesheet" href="/_css/light.css" media="(prefers-color-scheme: no-preference), (prefers-color-scheme: light)">
```

Comment:
Once we remove the above `<script>`, we can simplify things further by moving the contents of `light.css` back into `main.css`, and then only conditionally loading `dark.css` when dark mode is enabled.

Reply:
Should we load these below `main.css`?

Reply:
I've added a suggestion below on just keeping it all in one file. IMO that would be simpler (including for fallback logic) and even faster than separate requests given the size of these files.

Reply:
The less files the better!